pkg/common/version/version.go

```diff
@@ -16,7 +16,13 @@ func Version() string {
 		// Metrics systems sometimes automatically reject values with full SHA1 hashes
 		// since they appear to be high cardinality.
 		// For our purposes, the first 7 chars of the SHA should be enough.
-		githashShort := githash[:7]
+		var githashShort string
+		if githash == "" {
+			githashShort = ""
+		} else {
+			githashShort = githash[:7]
+		}
+
 		return fmt.Sprintf("%s-dev-%s", Base, githashShort)
```

Comment:
Won't this leave a trailing `-` if `githash == ""`?

Reply:
Yes, resolved by changing the default value to "unk".